File: python/타겟 넘버.py
answer = 0


def dfs(cur, value,  numList, t, totalSize):

    global answer

    if cur == totalSize:

        if value == t:
            answer += 1
        return

    dfs(cur+1, value-numList[cur],  numList, t, totalSize)

    dfs(cur+1, value+numList[cur],  numList, t, totalSize)


def solution(numbers, target):

    size = len(numbers)

    dfs(0, 0,  numbers, target, size)

    return answer

File:  cpp/타겟 넘버.cpp
#include <string>
#include <vector>


using namespace std;

int answer2=0;
int target2;

void go(vector<int> numbers, int start, int cnt, int dab){
    if(cnt == numbers.size()){
        if(target2==dab){
            answer2++;
        }
        return;
    }



        go(numbers, start+1, cnt+1, dab+numbers[start]);
        go(numbers, start+1, cnt+1, dab-numbers[start]);


}

int solution(vector<int> numbers, int target) {
    int answer = 0;

    target2= target;
    go(numbers, 0, 0, 0);

    answer=answer2;

    return answer;
}